langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/FromString.java

```diff
@@ -37,6 +40,11 @@ public class FromString {
 
     public static Object fromString(BString string) {
         try {
+            if (string.length() == 0) {
+                throw BLangExceptionHelper.getRuntimeException(BallerinaErrorReasons.REG_EXP_PARSING_ERROR,
+                        RuntimeErrors.EMPTY_STRING);
+            }
+
             return TypeConverter.stringToRegExp(string.getValue());
         } catch (BError bError) {
```

Comment:
You should be able to remove this catch

Reply:
Wouldn't removing it cause a compilation error due to a `BError` being thrown at `ypeConverter.stringToRegExp()` for invalid regexp syntax?